TFNetworkLayer.py

```diff
@@ -1966,7 +1998,11 @@ class SoftmaxLayer(LinearLayer):
   layer_class = "softmax"
 
   def __init__(self, activation="softmax", **kwargs):
-    super(SoftmaxLayer, self).__init__(activation=activation, **kwargs)
+    super(SoftmaxLayer, self).__init__(activation=activation, out_feature_dim_axis=-1, **kwargs)
```

Comment:
I still don't understand why that is needed.
Or to say it differently: It's inconsistent. Why should the `SoftmaxLayer` have a different default than the `LinearLayer`? There is no reason for it. So don't do that.